### PR TITLE
Set dev server listen host and port via environment variables

### LINKED ISSUE #767: Dynamic port for webpacker-dev-server
Webpacker configuration file is [extensively reloaded](https://github.com/rails/webpacker/blob/master/lib/webpacker/railtie.rb#L38) and it seems there is no straight forward way to dynamically set dev server port, e.g., using Foreman to start Rails and webpacker-dev-server. Any suggestions how to do that?

### PR DESCRIPTION
**Pull request by monai**
Resolve #767.

### CODE REVIEW

**Comment by javan** [comment]
If we're going to support `ENV` vars, I think we should convert all of `bin/webpack-dev-server`s `ARGV`s to them.

File: lib/webpacker/dev_server.rb [hunk 1]
```diff
@@ -21,11 +21,11 @@ def hot_module_replacing?
   end
 
   def host
-    fetch(:host)
+    ENV["WEBPACK_DEV_SERVER_LISTEN_HOST"] || fetch(:host)
```
Comment (javan) [inline lib/webpacker/dev_server.rb]: These should all match their method names, e.g. `ENV["WEBPACK_DEV_SERVER_HOST"]`

**Comment by gauravtiwari** [comment]
Yepp and may be best to get rid of `ARGV` support for dev server binstub and add important ones to webpacker.yml itself? Obviously for run time options (host etc.) we can have ENV variables.

**Comment by javan** [comment]
Yeah, I'd like to drop `ARGV` support in favor of `ENV`. Maybe like this?
```diff
--- a/lib/webpacker/dev_server.rb
+++ b/lib/webpacker/dev_server.rb
@@ -42,7 +42,7 @@ class Webpacker::DevServer
 
   private
     def fetch(key)
-      config.dev_server.fetch(key, defaults[key])
+      ENV["WEBPACK_DEV_SERVER_#{key.upcase}"] || config.dev_server.fetch(key, defaults[key])
     end
```

**Comment by gauravtiwari** [comment]
Yepp looks great 👍

**Comment by monai** [comment]
And in binstub we can take all `WEBPACK_DEV_SERVER_` variables and merge them with config file. I'm going to update PR in few days.

**Comment by gauravtiwari** [comment]
@monai This involves a few more changes so I am going to make another PR that covers everything 👍

**Comment by gauravtiwari** [comment]
Closing in favour of #843

**Pull request closed without merging**